sky/serve/serve_utils.py

```diff
@@ -719,6 +722,139 @@ def _get_replica_status() -> serve_state.ReplicaStatus:
+    terminated_replicas = [
+        info for info in service_record['replica_info']
+        if info['status'] in serve_state.ReplicaStatus.terminal_statuses()
+    ]
+    for replica in terminated_replicas:
+        replica_id = replica['replica_id']
+        log_file_name = generate_replica_log_file_name(service_name, replica_id)
+        if has_valid_replica_id(log_file_name, target_replica_id):
+            if os.path.exists(log_file_name):
+                shutil.copy(log_file_name, dir_for_download)
+
+    # These copies of logs may still be continuously updating.
+    # We need to synchronize the latest log data from the remote server.
+    for replica in service_record['replica_info']:
+        replica_id = replica['replica_id']
+        launch_log_file = generate_replica_launch_log_file_name(
+            service_name, replica_id)
+
+        # Check if launch log file exists and has correct replica_id.
+        if not (os.path.exists(launch_log_file) and
+                has_valid_replica_id(launch_log_file, target_replica_id)):
+            logger.error(
+                f'{colorama.Fore.RED}Path {launch_log_file} does not exist, '
+                f'or replica_id {replica_id} is invalid.'
+                f'{colorama.Style.RESET_ALL}')
+            continue
+
+        # Copy launch log file into the download directory.
+        new_replica_log_file = os.path.join(dir_for_download,
+                                            f'replica_{replica_id}.log')
+        shutil.copy(launch_log_file, new_replica_log_file)
+
+        # Sync down job logs for replica.
+        if replica['status'] == serve_state.ReplicaStatus.PROVISIONING:
+            continue
+        logger.info(f'Syncing down logs for replica {replica_id}...')
+        backend = backends.CloudVmRayBackend()
+        handle = global_user_state.get_handle_from_cluster_name(
+            replica['cluster_name'])
+        if handle is None:
+            logger.error(f'Cannot find cluster {replica["cluster_name"]} for '
+                         f'replica {replica_id} in the cluster table. '
+                         'Skipping syncing down job logs.')
+            continue
+        assert isinstance(handle, backends.CloudVmRayResourceHandle)
+
+        # Set up local directory for replica job logs.
+        replica_job_logs_dir = os.path.join(skylet_constants.SKY_LOGS_DIRECTORY,
+                                            'replica_jobs')
+        os.makedirs(replica_job_logs_dir, exist_ok=True)
+        job_log_file_name = None
+        try:
+            # Try to sync down the logs for the replica.
+            log_dirs = backend.sync_down_logs(handle,
+                                              job_ids=None,
+                                              local_dir=replica_job_logs_dir)
+        except exceptions.CommandError as e:
+            logger.info('Failed to download the logs: '
+                        f'{common_utils.format_exception(e)}')
```

Comment:
Lets directly raise those errors. we should not transparently let the error happen